numpy/random/mtrand/mtrand.pyx

```diff
@@ -35,6 +33,17 @@ cdef extern from "math.h":
     double sin(double x)
     double cos(double x)
 
+# Put initarray first since it includes the NPY_NO_DEPRECATED_API definition
+cdef extern from "Python.h":
+    ctypedef int Py_intptr_t
```

Comment:
Is the fact that this is missing from the `Python.pxi` an upstream bug?

Reply:
which `Python.pxi`? The `cdef extern from "Python.h"` tells Cython to add `#include Python.h` at this point in the output C file, and promises that that header file (or one of the headers included by it) will include a `typedef int Py_intptr_t` statement.

Reply:
[this one](https://github.com/numpy/numpy/pull/12284/files#diff-70d0fdd5755b6d140e4f09d9de3da144R24) - my assumption was that this header already includes all of the `cdef externs` that match to `Python.h`. Is that not true?